app/controllers/emails_controller.rb

```diff
@@ -1,11 +1,87 @@
 class EmailsController < ApplicationController
+
+  def show
+    authorize! :send_email, Email
+    @event = Event.find(params[:event_id])
+
+    @templates = get_templates
+    @addresses = @event.email_addresses_of_type(get_status)
+
+    @email = Email.new(hide_recipients: true, reply_to: 'workshop.portal@hpi.de', recipients: @addresses,
+                       subject: '', content: '')
+    render :email
+  end
+
+  def submit
+    authorize! :send_email, Email
+    if send?
+      send_email
+    elsif save_template?
+      save_template
+    end
+  end
+
+
+  private
+
   def send_email
     @email = Email.new(email_params)
-    Mailer.send_generic_email(@email.hide_recipients, @email.recipients, @email.reply_to, @email.subject, @email.content)
-    redirect_to :events, notice: t('.sending_successful')
+    if @email.valid?
+      Mailer.send_generic_email(@email.hide_recipients, @email.recipients, @email.reply_to, @email.subject, @email.content)
+
+      @event = Event.find(params[:event_id])
+      @event.lock_application_status
+
+      redirect_to @event, notice: t('.sending_successful')
+    else
+      @event = Event.find(params[:event_id])
```

Comment:
I think its okay to set @event again, its a private method and might be called from other methods than show later on.
But I agree to move it before the if